### ISSUE TITLE
Cisco 4500x CPU graph not showing data since Feb 16

### ISSUE THREAD

**Issue by kdohBNY** [librenms/librenms#8559]
**DO NOT DELETE THIS INFORMATION.**

> **Please read this information carefully.**

**GitHub issues is for known/validated bugs, please do not post issues asking for help or how to do X, Y or Z.** 

**You can use our [Discord server](https://t.libren.ms/discord) to ask questions or our [community site](https://community.librenms.org).**

**If you have a feature request/changes please post this on our [community site](https://community.librenms.org/c/feature-requests).**

Please confirm each of the sections below by putting an `x` in the box like `[x]`.

- [x] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [x] Please include all of the information between the `====================================` section of `./validate.php` which you can run from the cli.
- [x] Please provide **ALL** info asked for [here](http://docs.librenms.org/Support/FAQ/#faq20).
- [x] Please provide as much detail as possible.
- [x] Please do NOT post more than 10 lines of debug information here, use a pastebin service or GitHub Gists.

Love librenms? Please consider supporting our collective:
👉  https://t.libren.ms/donations

====================================
Component | Version
--------- | -------
LibreNMS  | 1.38-48-gfbbc257
DB Schema | 247
PHP       | 7.0.28-0ubuntu0.16.04.1
MySQL     | 10.0.34-MariaDB-0ubuntu0.16.04.1
RRDTool   | 1.5.5
SNMP      | NET-SNMP 5.7.3
====================================

Discovery - https://p.libren.ms/view/b2bd6f5e
Poller - https://p.libren.ms/view/bdee0893
[iosxe.snmpwalk.txt](https://github.com/librenms/librenms/files/1908345/iosxe.snmpwalk.txt)

Starting at midnight on Feb. 16 none of my Cisco 4500x switches (VSS or single) are showing any output on their CPU graphs. Other cisco devices and other platforms seem to not have any issue so I'm thinking it's something specific to the 4500x platform. The only thing I can think of is maybe a LibreNMS update that went out at that time may have broken it? Everything else seems to be graphing properly and when I do the "show rrd command" it says it's ok.

Examples showing CPU stopped graphing at the same time. 
![575_cpu_graph](https://user-images.githubusercontent.com/28956529/38749313-28c412cc-3f1f-11e8-94a9-81104d5af428.PNG)
![lyn_cpu_graph](https://user-images.githubusercontent.com/28956529/38749316-2b15cce6-3f1f-11e8-8848-1e414f50cc63.PNG)
![colo_cpu_graph](https://user-images.githubusercontent.com/28956529/38749317-2cdf3da0-3f1f-11e8-8ad8-ffb00b511dd5.PNG)

Please let me know if there is anything else you need from me or if I can provide anything else to help troubleshoot this.

Thanks guys!

**Comment by laf** [comment]
The walk you provided shows a CPU graph fine, 24% usage.

The poller and discovery output are empty, most likely because the hostname used is wrong.

From our commits, it doesn't look like anything CPU related was merged the 15th - 18th Feb so unless you manually updated on that date to a specific commit I can't see what's up.

However, please open a thread on the community forum, GitHub is for known or validated bugs as per the original template.

**Comment by murrant** [comment]
@kdohBNY The discovery shows the data and the correct oids, but when we snmpget the data in discovery, it doesn't return anything.  Could be an issue with the device.  Try running the snmpget command you see in the poller manually.

**Comment by kdohBNY** [comment]
Hi @murrant 
The snmpget on 11000.0 gives me an error interestingly enough but not on 11000.1:
command: snmpget -v2c -c netassess -OQUsetX -m CISCO-PROCESS-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/cisco udp:HOSTNAME:161 .1.3.6.1.4.1.9.9.109.1.1.2.1.5.11000.0

returns: .1.3.6.1.4.1.9.9.109.1.1.2.1.5.11000.0: Unknown Object Identifier (Index out of range: 0 (cpmCoreIndex))

command: snmpget -v2c -c netassess -OQUsetX -m CISCO-PROCESS-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/cisco udp:172.30.2.1:161 .1.3.6.1.4.1.9.9.109.1.1.2.1.5.11000.1

returns: cpmCore5min[11000][1] = 6

If I do a snmpwalk on 11000 it gives me both:

command: snmpwalk -v2c -c netassess -OQUsetX -m CISCO-PROCESS-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/cisco udp:172.30.2.1:161 .1.3.6.1.4.1.9.9.109.1.1.2.1.5.11000

cpmCore5min[11000][0] = 11
cpmCore5min[11000][1] = 6

I would have thought that querying the 11000.0 would return the 'cpmCore5min[11000][0] = 11' output from the snmpwalk. Am I missing something obvious?

Sorry to be a bother but I'm still learning all this myself.

**Comment by laf** [comment]
If you can't snmpget the valid OID then that would seem like a device issue - did you do firmware updates around that time?

**Comment by murrant** [comment]
Yeah, snmpget should return the data as you expect.  Looks like a device bug,  you may want to check for firmware updates.

**Comment by kdohBNY** [comment]
Hey guys - Thanks again for all the help with troubleshooting the issue. Looks like the issue was due to the local snmp installation on my server. I was able to get it working again by reinstalling net-snmp and restarting/reconfiguring the service. As a side note, I'm still getting the "Index out of range" error when doing the snmpget command but it appears to be graphing properly again post restart of the service. 

Thanks again ~Cheers!
